### PR TITLE
Strerror & GetMidiEvents

### PR DESCRIPTION
**Pull request by esqilin**
This is my first pull request on github; after reading the guidelines I realize that I should have created one or two new branches.

I created an extra file with function _Strerror_ for translating JACK status values into Go errors.
Completely unrelated, the second commit adds the function _GetMidiEvents_ which returns a slice of MidiData pointers. Maybe it is too high level, maybe it should come in a separate subpackage because it necessarily imports _jack/midiport.h_.

Thanks for the implementation, I also wrote a wrapper and realized that it became way to high level so I switched my projects to your go-jack.

### CODE REVIEW

**Comment by xthexder** [comment]
This looks pretty good, though if we add midi support, it would be nice to have full support rather than just input/read only.

If you'd like you could put something together for midi writes, or I could take a look at it. I haven't really used Jack with midi before so it could be interesting.

**Comment by esqilin** [comment]
Cool, I'll add full support and an example. Easter Holidays are over 
though and I have three excursions coming up so I won't be able to do it 
within the next days.
Thanks for looking at it!

**Comment by esqilin** [comment]
Done. Maybe example/Makefile needs to be adjusted.

File: errors.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	case ShmFailure:
+		msg = "unable to access shared memory"
+	case VersionError:
+		msg = "client's protocol version does not match"
+	case BackendError:
+		msg = "backend error"
+	case ClientZombie:
+		msg = "client zombie"
+	case C.EEXIST:
+		msg = "the connection is already made"
```
Comment (xthexder) [inline errors.go:41]: Small nitpick, but this should probably include some of the other `sys/errno.h` error codes here. Some of the ones the midi library can return:

```
ENODATA if buffer is empty
ENOBUFS if there's not enough space in buffer for event
```

**Comment by xthexder** [comment]
Just the one thing, but other than that this looks good :+1: 
I'll give it a try tonight and merge it when I'm home from work.

**Comment by sstark** [comment]
Hello, any news if this is going to be merged? I am looking for an alternative to portmidi and jack looks promising. In my project I am only writing midi out. Just it looks to me as if jack will take quite some time to grasp, and before I dive into it I'd like to know if I will be able to use it from go at all. Thank you anyway for making these bindings!

**Comment by xthexder** [comment]
Sorry about this, this kind of got buried under other work I've been doing, this looks good enough to merge, though I haven't been able to test it myself yet.

**Pull request merged**